src/plots/cartesian/position_defaults.js

```diff
@@ -27,6 +27,9 @@ module.exports = function handlePositionDefaults(containerIn, containerOut, coer
     dfltAnchor = dfltAnchor || (isNumeric(containerIn.position) ? 'free' : (counterAxes[0] || 'free'));
     dfltSide = dfltSide || (letter === 'x' ? 'bottom' : 'left');
     dfltPosition = dfltPosition || 0;
+    dfltAutoshift = dfltAutoshift || false;
+    dfltShift = dfltShift || 0;
+    dfltAutomargin = dfltAutomargin || false;
```

Comment:
You don't need `dfltAutoshift` at all - if an attribute has a static `dflt` you shouldn't even use the second argument to `coerce`. And `dfltShift` and `dfltAutomargin` don't have any prior logic so should just be `= 0` and `= false` respectively.